<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>







  
  
  
  
  
  
  
  <meta content="text/html; charset=ISO-8859-1" http-equiv="content-type">







  
  
  
  
  
  
  
  <title>customQuality</title>
</head>


<body>






<h1 align="center">Custom Quality Control for Spotted Arrays</h1>












<p align="center">April 18, 2008</p>











<div style="text-align: center;">Agnes Paquet<sup>1</sup>, Andrea Barczak<sup>1</sup>, (Jean) Yee Hwa
Yang<sup>2</sup> <br>












<br>












1. Department of Medicine, Functional Genomics Core Facility,
University of California, San Francisco<br>












paquetagnes@yahoo.com<br>












2. School of Mathematics and Statistics, University of Sydney, Australia<br>






</div>












<a href="http://arrays.ucsf.edu/analysis/arrayquality.html"><span style="color: rgb(255, 0, 0);"></span></a><br>






<br>






<h1>Content</h1>






<span style="font-family: Courier New,Courier,monospace;">ArrayQuality</span>provides
a flexible framework for assessment of hybridization quality. Many of
the functions&nbsp;can be customized to better suit user's
needs. For example, <span style="font-family: courier new,courier,monospace;">arrayQuality </span>is 
currently using look-up tables adapted to hybridizations performed in the 
Functional Genomics Core Facility at UCSF.&nbsp; Depending on your data, you may find 
that the probes defined as controls in <span style="font-family: courier new,courier,monospace;">arrayQuality </span>are not 
present on your array, leading to NAs in the comparative boxplot, or you may be 
working with a genome for which we are not providing references. In this section, we will describe how to:<br>






<ul>






  <li>Define your own controls&nbsp;in <span style="font-family: Courier New,Courier,monospace;">gpQuality</span></li>






  <li><span style="font-family: Courier New,Courier,monospace;"></span>Use your own collection of good quality slides for quantitative assessment of the comparative boxplot part of <span style="font-family: Courier New,Courier,monospace;">gpQuality</span></li>






  <li>Define a new quantitative quality measure in the comparative boxplot<span style="font-weight: bold;"></span></li>






</ul>





<span style="font-weight: bold;"></span>All example code will be
provided for GenePix data format, but the same functionalities are
available for Agilent or Spot data format.&nbsp; <br>





<span style="font-weight: bold;"></span><br>






<span style="font-weight: bold;"><span style="color: rgb(255, 0, 0);"><span style="color: rgb(0, 0, 0);"></span>Warning</span>:</span>
Modifying the default settings requires a good understanding
of two-color array design and some programming experience in R. Only
more advanced users should use the functionalities described in
this section. Other users should start by first using&nbsp;default
functions for general hybridization, as described in the basicQuality
user's guide, available from the main online help page.<br>






<br>






<h1>1. How to specify your own set of controls or spot types in <span style="font-family: Courier New,Courier,monospace;">gpQuality</span></h1>






<span style="font-family: courier new,courier,monospace;"></span><span style="font-family: courier new,courier,monospace;">gpQuality </span>has 
several arguments that you can modify <span style="font-family: times new roman,times,serif;">in order</span><span style="font-family: courier new,courier,monospace;"><span style="font-family: times new roman,times,serif;"> to use your own spot types or 
your own collection of good slides. gpQuality arguments are listed 
below:</span></span><br>






<span style="font-family: courier new,courier,monospace;"><br>






gpQuality(fnames = NULL, 
path = ".", organism = c("Mm", "Hs"),<br>






&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp; compBoxplot = TRUE, reference 
= NULL, <br>






&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp; controlMatrix = controlCode, controlId = c("ID", "Name"), 
<br>






&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; output = FALSE, resdir =".", dev= "png", DEBUG = 
FALSE,...)<br>






<span style="font-family: times new roman,times,serif;"><br>






<br>






To 
use your own set of spot types (i.e. controls...): you will need to change <span style="font-family: courier new,courier,monospace;">controlMatrix</span> and/or 
<span style="font-family: courier new,courier,monospace;">controlId</span>.<br>






<br>






</span></span>The spot types currenlty used in arrayQuality are defined in 
a 2-column matrix called <span style="font-weight: bold;">controlCode</span>. <br>






<br>







<table style="margin-left: auto; width: 300px; margin-right: auto; text-align: left;" border="1" cellpadding="2" cellspacing="2">







  <tbody>







    <tr>







      <td style="vertical-align: top;">Pattern<br>






      </td>







      <td style="vertical-align: top;">Name<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">Buffer<br>






      </td>







      <td style="vertical-align: top;">Buffer<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">Empty<br>






      </td>







      <td style="vertical-align: top;">Empty<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">EMPTY<br>






      </td>







      <td style="vertical-align: top;">Empty<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">AT<br>






      </td>







      <td style="vertical-align: top;">Negative<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">M200009348<br>






      </td>







      <td style="vertical-align: top;">Positive<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">M200003425<br>






      </td>







      <td style="vertical-align: top;">Positive<br>






      </td>






    </tr>







    <tr>







      <td style="vertical-align: top;">NLG<br>






      </td>







      <td style="vertical-align: top;">con<br>






      </td>






    </tr>






  
  
  
  
  
  
  </tbody>
</table>






<br>







<div style="text-align: center;">Table 1: Examples of controls used in 
arrayQuality<br>






</div>






<br>






<br>






To define your own spot types, you will need to 
replace the default values in controlCode with the controls present on your arrays. The easiest way to 
do it is to create a&nbsp; tab-delimited text file named SpotTypes.txt, and read it 
into arrayQuality using the function <span style="font-family: Courier New,Courier,monospace;">readcontrolCode</span>. It is also possible to 
create a new controlCode matrix directly.<br>






<br>






<h2>1.1 If you want to use a Spot 
Types file:</h2>






A spot types file is a tab-delimited text file which allows 
you to identify different types of spots from the gene list. It should contain 
at least a column named SpotType where all different spot types are listed and 
one or more other columns, which should have the same names as columns in the 
GAL file, containing patterns or regular expressions sufficient to identify the 
spot-type. For more information, you can refer to the <span style="font-family: courier new,courier,monospace;">limma</span> package 
user's guide. <br>






<br>






<span style="font-weight: bold;">Warning:</span> You will need 
to include a spot type named <span style="font-weight: bold;">probes</span>!!<br>






<br>






Below is an example of spot 
types files for the swirl dataset. In this case there are only two types of 
spots, probes and controls.<br>






<br>







<div style="text-align: center;"><img title="" style="width: 417px; height: 214px;" alt="example of spot types file" src="images/mySpotTypes.png"><br>






</div>







<div style="text-align: center;"><br>






</div>







<div style="text-align: center;">Example of spot types file<br>






</div>






<br>






<br>






To 
read the new spot types in arrayQuality:<br>






<br>






- Create your spot types 
file.<br>






- Find which column of the file contains probes identification for each 
type. In the example Figure 8, it is the "ID" column. You will need to pass this 
column name as argument at the next step.<br>






- Read the spot types files using 
the readcontrolCode function. <br>







<p class="MsoNormal" style="font-family: courier new,courier,monospace; text-align: justify;">&gt; 
controlCode &lt;- readcontrolCode(file=&rdquo;mySpotTypes.txt&rdquo;, 
controlId="ID")<br>






</p>







<p class="MsoNormal" style="text-align: justify;">- Find which column of the gpr 
file can be used to identify your new spot types. It is typically the "ID" or 
the "Name" column.<br>






- To generate both types of plots: call <span style="font-family: courier new,courier,monospace;">gpQuality</span> specifying 
your new controlCode matrix in <span style="font-family: courier new,courier,monospace;">controlMatrix</span> and 
which column is used to define your spot types in <span style="font-family: courier new,courier,monospace;">controlId</span>.<br>






</p>







<p class="MsoNormal" style="font-family: courier new,courier,monospace; text-align: justify;">&gt; 
result &lt;- gpQuality(controlMatrix = controlCode, controlId=&rdquo;Id&rdquo;)</p>






<br>






<h2>1.2 If 
you want to create a new <span style="font-family: courier new,courier,monospace;">controlCode</span> matrix 
directly</h2>






You will need to create another <span style="font-family: courier new,courier,monospace;">controlCode </span>table 
containing two columns as well, and then overwrite the default <span style="font-family: courier new,courier,monospace;">controlCode </span>loaded 
with <span style="font-family: courier new,courier,monospace;">arrayQuality</span>. The <span style="font-family: Courier New,Courier,monospace;">controlCode</span> matrix must&nbsp; have 2 columns:<br>






&nbsp;&nbsp;&nbsp; - 
A column named "Pattern" containing your control IDs<br>






&nbsp;&nbsp;&nbsp; - A column named 
"Name", describing what king of control is each probe (in particular what are 
Positive, Negative, Empty controls)<br>






<br>






You can do it by creating a tab 
delimited text file and read it in R after loading <span style="font-family: courier new,courier,monospace;">arrayQuality</span>:<br>






<br>






<span style="font-family: courier new,courier,monospace;">&gt; 
library(arrayQuality)<br>






&gt; mycontrolCode &lt;- 
as.matrix(read.table("mycontrolCode.txt", sep="\t",</span><br style="font-family: courier new,courier,monospace;">






<span style="font-family: courier new,courier,monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
header=TRUE, quote="\"", fill=TRUE)))<br>






<br>






<span style="font-family: times new roman,times,serif;">Then, pass your new matrix as 
argument when calling gpQuality. You can specify which column of the gpr file 
contains probes identifiers in the controlId arguments (typically, it would be 
"Id" or "Name").</span><br>






&gt; results &lt;- gpQuality(controlMatrix = 
mycontrolCode, controlId = "ID")<br style="font-family: courier new,courier,monospace;">






</span><br>











<span style="font-family: courier new,courier,monospace;"><span style="font-family: times new roman,times,serif;"><br>





</span></span>
<h1><span style="font-family: courier new,courier,monospace;"><span style="font-family: times new roman,times,serif;">
2. How to use your own collection of good quality reference slides</span></span></h1>





<span style="font-family: times new roman,times,serif;">T</span><span style="font-family: courier new,courier,monospace;"><span style="font-family: times new roman,times,serif;">he
comparative boxplot generated as part of the general hybridization
diagnostic plot can be used to detect outlier arrays within in a large
dataset or if you want to study hybridization quality for other genomes.
This is done by comparing some quantitative statistics from each array
to a range of values corresponding to good quality arrays, which are determined over a collection of "good quality" arrays
selected from the same dataset. &nbsp;To use your own collection
of good slides: you will need to modify a look-up table named&nbsp; <span style="font-family: courier new,courier,monospace;">reference</span> passed in the arguments of <span style="font-family: Courier New,Courier,monospace;">gpQuality</span>.</span></span><br>





<span style="font-family: courier new,courier,monospace;">
</span><span style="font-weight: bold;"></span><br>






<span style="font-weight: bold;"></span>To generate your 
own <span style="font-family: Courier New,Courier,monospace;">reference</span>:<br>






<br>






&nbsp;&nbsp;&nbsp; 1. Gather the slides of "good" quality you would like 
to use as reference in a directory, for example "MyReferences". Slides can be 
from different print-runs.<br>






<br>






&nbsp;&nbsp;&nbsp; 2. Change your R working directory to 
"MyReferences"<br>






<br>






&nbsp;&nbsp;&nbsp; 3. Load arrayQuality 
package by typing <span style="font-family: courier new;">library(arrayQuality)</span> in R<br>






<br>






&nbsp;&nbsp;&nbsp; 4. 
Create your reference quality measures by typing:<br>






&nbsp;&nbsp;&nbsp; <span style="font-family: courier new;">&gt; myReference &lt;-&nbsp; 
globalQuality()</span><br style="font-family: courier new;">






<br>






&nbsp;&nbsp;&nbsp; 5. Change R 
working directory to the directory containing slides&nbsp;you would like to test, as 
described. You can only compare slides from the same print-run 
here. If you have an experiment using 2 print-runs, you will need to 
run gpQuality 2 times.<br>






<br>






&nbsp;&nbsp;&nbsp; 6. Run gpQuality using the reference 
measures and the scaling table you have generated:<br>






&nbsp;&nbsp;&nbsp; <span style="font-family: courier new;">&gt; results &lt;- gpQuality(reference = 
myReference)</span><br>






<br>






&nbsp;&nbsp;&nbsp; Other <span style="font-family: courier new;">gpQuality </span>arguments described above can 
also be applied here.<br>







<p class="MsoCaption" style="text-align: left;"></p>





<h1>3. How to define your own quantitative measures for the comparative boxplot</h1>



We have selected a set of 15 measures that can be used to verify
quantitatively the quality of an array. These measures are described in
detail in the basicQuality guide, that can be accessd from the main
help page of the package. This part can be customized &nbsp;a step
further, as it is possible to&nbsp;define your own set of quantitative
measures. &nbsp;The quantitative boxplot is generated as follows:<br>



<br>



<ol>



  <li>Read in the array data from the image analysis software input. This can be done by calling the function <span style="font-family: Courier New,Courier,monospace;">readGPR</span>, which stores specific columns from the input file in a list.</li>



  <li>Calculate the quantitative measures and statistics you need for
quality assessment. The is done one slide at a time using the function <span style="font-family: Courier New,Courier,monospace;">slideQuality</span>. You will need to create your own <span style="font-family: Courier New,Courier,monospace;">slideQuality</span> function if you prefer to use your own set of quantitative measures.</li>



  <li>Select a collection of "good quality" arrays to use as reference.</li>



  <li>Estimate the range of each measure corresponding to "good
quality" arrays by repeating steps 1 and 2 for each array in your
collection. Save as your new <span style="font-family: Courier New,Courier,monospace;">reference</span> (see section 2 above for details).</li>



  <li>Draw the comparative boxplot for each slide using your new <span style="font-family: Courier New,Courier,monospace;">reference</span>.</li>



  <li>Estimate the QC score for your array using the <span style="font-family: Courier New,Courier,monospace;">qcScore</span> function.</li>



</ol>





The new slideQuality function must call the same argument as
slidequality, and return a matrix. Here is an example R code for such a
function.<br>



The <span style="font-family: Courier New,Courier,monospace;">gprData</span> argument is the result you get from running <span style="font-family: Courier New,Courier,monospace;">readGPR</span> on your array. <span style="font-family: Courier New,Courier,monospace;">readGPR</span>
reads in specific columns from the gpr file only, and returns the
results in a list. Depending on your needs, you may need to create your
own reading function as well. Then, you can use the elements from <span style="font-family: Courier New,Courier,monospace;">gprData</span>
to calculate the QC measures you are interested in and return them in a
matrix. This example function returns 4 quality measures: the range of
red forground, the range of green foreground, the median spot area and
the spot radius. &nbsp; <br>



<br>



<span style="font-family: Courier New,Courier,monospace;">
mySlideQuality &lt;- function(gprData = NULL, controlMatrix = controlCode, controlId = c("ID", "Name"), DEBUG = FALSE, ...) </span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



{</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; Rf &lt;- log.na(gprData[["RfMedian"]], 2)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; Gf &lt;- log.na(gprData[["GfMedian"]], 2)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; rRf &lt;- range(Rf, na.rm = TRUE)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; rangeRf &lt;- rRf[2] - rRf[1]</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; rGf &lt;- range(Gf, na.rm = TRUE)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; rangeGf &lt;- rGf[2] - rGf[1] </span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; spotArea &lt;- median(gprData[["spotArea"]], na.rm = TRUE)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; spotRadius &lt;- round(sqrt(spotArea)/pi)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; sortedMeasures &lt;- c("range RF", "range GF", "spotArea", "spotRadius")</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; sortedRes &lt;- c(rangeRf, rangeGf, spotArea, spotRadius)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; numResult &lt;- as.matrix(sortedRes)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; rownames(numResult) &lt;- sortedMeasures</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; colnames(numResult) &lt;- gprData[["File"]]</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



&nbsp; return(numResult)</span><br style="font-family: Courier New,Courier,monospace;">



<span style="font-family: Courier New,Courier,monospace;">



}</span><br>







<br>



<h2>Example</h2>



1) Create mySlideQuality (as above)<br>


2) Run the following R code:<br>


<span style="font-family: Courier New,Courier,monospace;">




&gt; datadir &lt;- system.file("gprQCData", package="arrayQuality")</span><br style="font-family: Courier New,Courier,monospace;">


<span style="font-family: Courier New,Courier,monospace;">




&gt; gprData &lt;- readGPR(fnames="9Mm137.gpr", path=datadir)&nbsp; ## read gpr file</span><br style="font-family: Courier New,Courier,monospace;">


<span style="font-family: Courier New,Courier,monospace;">




&gt; res = cbind(mySlideQuality(gprData),mySlideQuality(gprData) + 2, mySlideQuality(gprData)+4) ## create dummy reference</span><br style="font-family: Courier New,Courier,monospace;">


<span style="font-family: Courier New,Courier,monospace;">




&gt; res1 = mySlideQuality(gprData) ## run the new function of the slide</span><br style="font-family: Courier New,Courier,monospace;">


<span style="font-family: Courier New,Courier,monospace;">




&gt; qualBoxplot(arrayQuality = res1, reference = res) ## draw the boxplot</span><br style="font-family: Courier New,Courier,monospace;">


<span style="font-family: Courier New,Courier,monospace;">




&gt; qcScore(arrayQuality = res1, reference = res) ## calculate QC score</span>
</body>
</html>
