<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta content="text/html; charset=iso-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>arrayQuality User Manual</title>
</head>
<body>
<h1 align="center">Quality Control for spotted arrays</h1>
<p align="center"> October 17, 2004<br>
Agnes Paquet<sup>1</sup>, Andrea Barczak<sup>1</sup>, (Jean) Yee Hwa
Yang<sup>1,2</sup> <br>
<br>
1. Department of Medicine, Functional Genomics Core Facility,
University of California, San Francisco<br>
apaquet@medsfgh.ucsf.edu<br>
2. Department of Medicine, University of California, San Francisco<br>
<a href="http://arrays.ucsf.edu/analysis/arrayquality.html">http://arrays.ucsf.edu/analysis/arrayquality.html</a><br>
<br>
</p>
<h1>1. Introduction to arrayQuality</h1>
<p align="justify">
ArrayQuality is a R package, available as part of Bioconductor,
designed to help assessing quality of spotted array experiments at
several stages of the microarray lifecycle. It provides reports
containing several plots and statistical measures that can help you
determine if your hybridizations and slides are of good quality. More
information about Bioconductor is available at <a
 href="http://www.bioconductor.org">http://www.bioconductor.org</a>. </p>
<p>This guide provides an introduction to microarray quality and a
description of the main functionnalities of the package. A full
description of the package is given by the individual function help
documents available from the R online help system. To access the online
help, type <font face="courier">help(package=limma)</font> at the R
prompt or else start the html help system using <font face="courier">help.start()</font>
or the Windows drop-down help menu.</p>
<h1>2. Installing arrayQuality</h1>
<h2>Requirements</h2>
<p>
ArrayQuality is a library for the R project, part of Bioconductor. You
will need to have R installed on your computer before installing
arrayQuality. For more information about R, see the R project at <a
 href="http://www.r-project.org">http://www.r-project.org</a>.
ArrayQuality can work on different files at the same time, <b>ONLY</b>
if they are from the <b>SAME print-run</b> (same GAL file). If you
want to generate quality reports of slides from different print-runs,
you need to place them in different folders, one for each print-run.<br>
</p>
<p><span style="text-decoration: underline; font-weight: bold;">Note</span>:
[October 29 2004]<br>
The functions maQualityPlots, gpQuality, spotQuality, agQuality require
an older version of the hexbin package. You can install it either from
Bioconductor release 1.4 or from <a
 href="http://arrays.ucsf.edu/software/">http://arrays.ucsf.edu/software/</a>.<br>
</p>
<h2>Installing arrayQuality</h2>
<p>ArrayQuality can be installed either from Bioconductor or from
the functional Genomics Core Facility web site at
http://arrays.ucsf.edu/software. The version from Bioconductor is
updated every 6 months. If
you would like to use a more recent version, you can obtain the latest
one from&nbsp; http://arrays.ucsf.edu/software or from the
developmental version of
Bioconductor.
</p>
<ul>
  <li><b>Automatic installation from Bioconductor</b><br>
Start R on your computer and make sure you are connected to the
internet. At the R prompt, type:<br>
    <font face="courier">library(reposTools)<br>
install.packages2("arrayQuality")</font> <br>
    <br>
If you are using RGUI (Windows):<br>
In the R menu, click on "Packages". In the drop-down menu, select
"Install package(s) from Bioconductor". Then, select arrayQuality in
the list and click "OK".<br>
    <br>
  </li>
  <br>
  <li><b>Automatic installation from the Core Facility web site</b><br>
Start R on your computer and make sure you are connected to the
internet.<br>
    <br>
You can&nbsp; install the latest version directly from the Functional
Genomics Core Facility web site. At the R prompt, type:<br>
    <font face="courier">install.packages("arrayQuality",
contriburl="http://arrays.ucsf.edu/software")</font> <br>
    <br>
If you do not have privileges on your computer to write to the R
library directory, for example if you are using a shared unix machine
and you are not superuser, you may need to type instead<br>
    <font face="courier">install.packages("arrayQuality",
contriburl="http://arrays.ucsf.edu/software" lib="myRlibdir")</font> <br>
    <br>
  </li>
  <li><b>To load arrayQuality:</b><br>
Type at the R prompt:<br>
    <font face="courier">library(arrayQuality)</font> <br>
    <br>
  </li>
  <li><b>Manual installation</b><br>
Download arrayQuality as a .zip file (Windows users) from Bioconductor
or our web site. Start R on your computer and make sure you are
connected to the internet.<br>
In the R console, click on "Packages".<br>
In the drop-down menu, select "Install package(s) from local zip
file..."<br>
Browse to the zip file you would like to install, and click "Open". </li>
</ul>
<br>
<br>
<h1>3. Quick starting guide to arrayQuality</h1>
<h2>General hybridization quality</h2>
<p>
This component of the package is aimed at verifying the performance of
your hybridization, given the good quality of the slide, before any
preprocessing steps or further quality assessment on individual spots.
Our package provides two kinds of quality control plots:<br>
</p>
<ul>
  <li> A qualitative quality control measurement as a diagnostic plot.
It is a quick visual way to determine hybridization quality gathering
information from several statistical tools. </li>
  <br>
  <br>
  <li>A more quantitative comparison of slide quality as a comparative
boxplot. We extract some statistical measures from the test slide and
we compare them against results of &#8220;good quality&#8221; slides to assess the
quality of the hybridization. </li>
</ul>
Diagnostic plots can be generated directly for output of the GenePix
(.gpr files), Spot (.spot files)and Agilent image processing software
packages.
Most arguments can be customized to match your own data: which probes
are used as controls, which columns of the image processing output file
are used to define your spot types... You can also specify your own
collection of good quality slides. For more details, please refer to
other sections of this manual.
<br>
<br>
<h3>To generate quality plots from image processing output files</h3>
We provide 3 main functions to generate quality plots:
<b><font face="courier">gpQuality(), spotQuality() and agQuality()</font></b>.
We will use gpQuality as an example, but the following can be directly
applied to spotQuality or agQuality.<br>
<br>
<ol>
  <li>Create a directory and move the image processing output files
(e.g. .gpr files) of the slides of interest to this directory. Make
sure that all files in the directory come from the SAME print-run (same
GAL file). </li>
  <br>
  <br>
  <li>Start R, and change R working directory to the one you have just
created. In the R menu, select File, then click on &#8220;Change dir&#8230;&#8221;.
Browse to your directory from the pop-up window, or enter it manually,
and click OK. To double check that you are in the correct directory: in
the File menu, click on &#8220;Display file(s)&#8230;&#8221;. </li>
  <br>
  <br>
  <li>To load the package in your R session: type<br>
    <font face="courier">library(arrayQuality)</font><br>
    <br>
If needed, you may have to install other required packages like marray,
limma, convert and hexbin. </li>
  <br>
  <br>
  <li>To generate both <b>diagnostic plots and comparative boxplots</b>
on all files in the directory, type:<br>
    <font face="courier">result &lt;- gpQuality(organism=&#8221;Mm&#8221;)</font> </li>
  <br>
  <br>
  <li>To generate <b>diagnostic plots</b> only, run:<br>
    <font face="courier">result &lt;- gpQuality(organism="Mm",
compBoxplot="FALSE")</font><br>
In this case, quantitative quality measures will not be calculated and
the HTML report will not be generated. </li>
  <br>
  <br>
  <li>To write down your quantitative quality measures and your
normalized data to a file: set output = TRUE when calling gpQuality:<br>
    <font face="courier">result &lt;- gpQuality(organism="Mm",
output=TRUE)</font> </li>
  <br>
  <br>
  <li>By default, arrayQuality uses print-tip loess normalization. If
you prefer to use another method, you can specify it in the norm
argument:<br>
    <font face="courier">result &lt;- gpQuality(norm="none")</font><br>
For more details about normalization methods, please refer to marray
package help. </li>
  <br>
  <br>
</ol>
<h3>To generate quality plots from marrayRaw or RGList objects</h3>
You can use the function <b><font face="courier">maQualityPlots()</font></b>
to generate diagnostic plots directly from your R object.
<ul>
  <li>To generate diagnostic plots: if rawdata is a marrayRaw or RGList
object, type in R:<br>
    <font face="courier">maQualityPlots(rawdata)</font> </li>
  <br>
  <br>
</ul>
<h3>Results</h3>
gpQuality, spotQuality and agQuality output:<br>
<ul>
  <li>In the working directory:
- two plots for each test slide (a diagnostic plot and a comparative
boxplot)
- a HTML quality report </li>
  <br>
  <br>
  <li>In the R global environment, the function returns a list of two
objects:
- A marrayRaw object describing all tested slides - A quality measures
matrix: this matrix contains all comparison measures values extracted
for each test slide. Each column of the matrix represents a different
slide. </li>
</ul>
<h2>Print-run quality control</h2>
<p>
This component of arrayQuality provides diagnostic plots for 9-mers
hybridization and Quality Control hybridization.<br>
<br>
</p>
<h3>9-mers analysis: PRv9mers()</h3>
This type of hybridization, which we term &#8220;9mers hyb&#8221;, uses small
oligonucleotides (random 9-mers) which will hybridize to each probe on
the arrays. This will help to determine the quality of spot morphology
as well as the presence or absence of spotted oligonucleotides. The
resulting data will be used to create a list of all missing spots.<br>
In the package, the graphical function to assess 9mers hybridization
quality is PRv9mers(). It runs using one single command line script. <br>
<br>
<h4>To generate diagnostic plots</h4>
<ol>
  <li>Copy all 9-mers hybridizations gpr files from the SAME print-run
(same GAL file) to a directory. </li>
  <br>
  <br>
  <li>Start R and change R working directory to the one containing your
gpr files (see above for more details). </li>
  <br>
  <br>
  <li>Load the package in your R session:<br>
    <font face="courier">library(arrayQuality)</font><br>
    <br>
If needed, you may have to install other required packages like marray,
limma, convert and hexbin. </li>
  <br>
  <br>
  <li>To generate diagnostic plots, type:<br>
    <font face="courier">PRv9mers(prname=&#8221;12Mm&#8221;)</font><br>
The prname argument represents the name of your print-run. For more
details about other arguments, please refer to the online manual. </li>
</ol>
<br>
<br>
<h4>Results</h4>
PRv9mers()provides the following results:
<ol>
  <li>Diagnostic plots as image in .png format for each tested slide </li>
  <li>An Excel file (typically named 9Mm9mer.xls, where 9Mm is the name
of your print-run, as passed to prname) containing for each spot on the
slide:
    <ul>
      <li>Name and ID of the spot </li>
      <li>The probability of being present or absent (p from EM
algorithm). If several files are tested together, you will have a
probability of being present/absent for each file. A spot is considered
absent if p &lt; 0.5. </li>
      <li>The average probability of being present or absent. </li>
      <li>The raw signal intensity (Signal column) or average raw
signal intensity if several files are tested together for each spot. </li>
    </ul>
  </li>
  <li>An Excel file (typically named 9MmMissing.xls, where 9Mm is the
name of
your print-run, as passed to prname) containing information on missing
probes only:
    <ul>
      <li> Name and ID of the spot </li>
      <li> The probability of being present or absent (p from EM
algorithm). If several files are tested together, you will have a
probability of being present/absent for each file. A spot is considered
absent if p &lt; 0.5. </li>
      <li>The average probability of being present or absent. </li>
      <li>The raw signal intensity (Signal column) or average raw
signal intensity if several files are tested together for each spot. </li>
    </ul>
  </li>
  <li>A text file (typically named 9MmQuickList.txt, where 9Mm is the
name of your print-run, as passed to prname) containing the missing
probes ids, each on a separate line. This file can be opened in any
word processing programd as well as being a "quick list" in Acuity
software (http://www.axon.com/gn_Acuity.html). </li>
</ol>
<br>
<br>
<h3>Quality Control hybridization: PRvQCHyb()</h3>
9-mers hybridizations help verify that oligonucleotides have been
spotted properly on the slides. The next print-run quality control step
will be:<br>
<ol>
  <li>Detect any difference in overall signal intensity compared to
other print-runs
    <ul>
      <li>70-mers oligonucleotides hybridizations</li>
      <li>Selection of several test slides to ensure that the same
quantity of material was spotted across the platter, as a print-run
will generate 255 slides using the same well for one probe. QCHybs are
performed using one slide from the beginning of the print, one from the
middle, one from the end (e.g. numbers 20,100 and 255 in the Functional
Genomics Core Facility). </li>
    </ul>
  </li>
  <li>Check if the GAL file was generated properly, i.e. check that no
error
was made with ordering or orientation of the plates during the print. </li>
  <li>Reproducibility:<br>
A good way to verify the quality of a new print is to hybridize known
samples to new slides. Then, we can compare signal intensity from the
new slides to existing data, and check that there is no loss in signal.
Log ratios (M) for known samples should be similar across print-runs.
Example of samples used for QCHybs includes Human Reference pool, Mouse
liver, Mouse lung, with dye swaps. </li>
</ol>
<h4>To generate diagnostic plots</h4>
<ol>
  <li>Create a directory and move the image processing output files
(.gpr files only) of the slides of interest to this directory. Make
sure that all files in the directory come from the SAME print-run (same
GAL file).</li>
  <br>
  <br>
  <li>Start R, and change R working directory to the one you have
created (see general hybridization paragraph above for more details)</li>
  <br>
  <br>
  <li>Load the package in your R session:<br>
    <font face="courier">library(arrayQuality)</font><br>
    <br>
If needed, you may have to install other required packages like marray,
limma, convert and hexbin. </li>
  <br>
  <br>
  <li>For QCHyb analysis, run the following command:<br>
    <br>
    <font face="courier">PRvQCHyb()</font><br>
    <br>
  </li>
</ol>
<h4>Results</h4>
PRvQCHyb() returns a diagnostic plot as an image in .png format for
each tested slide.
<h2>Introduction to microarray quality</h2>
<p>
A microarray experiment is composed of several steps, including
experimental design, sample preparation, and various statistical
analyses (figure 1). They are represented in the microarray lifecycle
below. As microarray technology is complex and sensitive, it is
important to assess the performance of each step before going to the
next one. In addition, this is also a good way to trace back the cycle
to understand potential causes for upstream problems.
<br>
<br>
</p>
<div align="center"><img src="images/lifecycle.png"
 alt="microarray experiment lifecycle" align="middle">
<br>
Figure 1: Microarray experiment lifecycle
<br>
<br>
</div>
For spotted array experiments, quality controls can be summarized into
4 steps:<br>
<ol>
  <li>Print quality<br>
    <ul>
      <li>9mers hybridization</li>
      <li>Quality Control hybridization</li>
    </ul>
  </li>
  <li>mRNA quality</li>
  <li>Array hybridization quality</li>
  <li>Spot quality</li>
</ol>
Each step must be performed in a sequential order, as represented in
Figure 2.<br>
<br>
<div align="center">
<img src="images/qualityDiagram.png"
 alt="Quality control steps for microarray experiment" align="middle">
<p align="center">Figure 2: Quality Control for spotted arrays
experiment</p>
</div>
Our package provides graphical tools to look at two of these
components: print-run quality and array hybridization quality.<br>
<ol type="A">
  <li>Print quality:<br>
    <br>
This component is highly tailored to the Shared Genomics Core Facility
at UCSF, but the framework can be adapted to other Core facilities or
laboratories printing their arrays. It is an essential component of a
printed array experiment, as any print pin, probe or slide surface
defect will affect the quality of hybridization to the slide, and this
can&#8217;t be fixed by statistics. Only prints that did pass the quality
control check will be used for actual hybridization. </li>
  <br>
  <br>
  <li>Hybridization quality:<br>
    <br>
This is a global assessment of the hybridization performance. It helps
determine for example any problem with the dyes, or uneven
hybridization. Then, once you have determined that your hybridization
is good, you can look at each individual spot quality, remove bad
spots, and perform statistical analysis. </li>
  <br>
  <br>
</ol>
<h2>3. Print-run quality control</h2>
<p>
When a print-run is completed, it is necessary to verify the quality of
the resulting arrays. This can be done by using two kinds of
hybridization to the new slides. The first type of hybridization, which
we term &#8220;9mers hyb&#8221;, uses small oligonucleotides (random 9-mers), which
will hybridize to each probe. This hybridization will help to determine
the quality of spot morphology as well as the presence or absence of
spotted oligonucleotides. The resulting data will be used to create a
list of all missing spots.<br>
<br>
The second type of hybridization, which we will term Quality Control
Hybridization (QCHyb), uses mRNA from predefined cell lines (e.g. liver
vs. pool, K562 vs. Human Universal Reference pool from Stratagene).
These hybridizations can be use as a more quantitative description of
the slides. The same comparison hybridizations are done for different
print-run, assessing their reproducibility. QCHybs are also used to
verify accuracy of GAL files, number of missing spots, binding
capacity, background signal intensity&#8230;<br>
<br>
The arrayQuality package provides specific tools to help assess quality
of slides for both 9-mers and QC hybridization.<br>
<br>
</p>
<h3>3.1 9-mers hybridizations</h3>
In the package, the graphical function to assess 9mers hybridization
quality is PRv9mers(). It runs using one single command line script. To
use it:<br>
<ul>
  <li>Copy all 9-mers hybridizations gpr files from the SAME print-run
(same GAL file) to a directory. </li>
  <li>Change R working directory to the one containing your gpr files
as described in section 1. </li>
  <li>Type:<br>
    <font face="courier">PRv9mers(prname=&#8221;12Mm&#8221;).</font> </li>
</ul>
The prname argument represents the name of your print-run. For more
details about other arguments, please refer to the online manual.<br>
<br>
<h4>Results</h4>
<font face="courier">PRv9mers()</font> provides the following results:<br>
<ol>
  <li>Diagnostic plots as image in .png format for each tested slide </li>
  <br>
  <br>
  <li>An Excel file (typically named 9Mm9mer.xls, where 9Mm is the name
of your print-run, as passed to prname) containing for each spot on the
slide:<br>
    <ul>
      <li>Name and ID of the spot</li>
      <li>The probability of being present or absent (p from EM
algorithm). If several files are tested together, you will have a
probability of being present/absent for each file.<br>
A spot is considered absent if p &lt; 0.5.</li>
      <li>The average probability of being present or absent.</li>
      <li>The raw signal intensity (Signal column) or average raw
signal intensity if several files are tested together for each spot.</li>
    </ul>
  </li>
  <li>An Excel file (typically named 9MmMissing.xls, where 9Mm is the
name of your print-run, as passed to prname) containing information on
missing probes only:<br>
    <br>
    <ul>
      <li>Name and ID of the spot</li>
      <li>The probability of being present or absent (p from EM
algorithm). If several files are tested together, you will have a
probability of being present/absent for each file.<br>
A spot is considered absent if p &lt; 0.5.</li>
      <li>The average probability of being present or absent.</li>
      <li>The raw signal intensity (Signal column) or average raw
signal intensity if several files are tested together for each spot.</li>
    </ul>
  </li>
  <br>
  <br>
  <li>A text file (typically named 9MmQuickList.txt, where 9Mm is the
name of your print-run, as passed to prname) containing the missing
probes ids, each on a separate line. This file can be opened in any
word processing program. </li>
</ol>
<h4>Description of the diagnostic plots</h4>
Figure 3 shows an example from a typical 9-mers hybridization. This
image is divided in 5 plots.<br>
<br>
<ol>
  <li>The first column (left) represents boxplots of log intensity, by
plates (top) and by print-tip group (bottom). In this example, you will
notice on the boxplot by plates (top left corner) that plates 44 and 48
have lower intensity and wider range than the others. Both plates
contain mostly empty controls, as designed by Operon. </li>
  <li>Central plot: spatial plot of intensity. This helps to locate
missing spots. The color scale reflects the signal intensity, the
darker the color of the plot, the stronger the signal. Missing spots
are represented in white. In Figure 3 spatial plot, top right corner
white spots come from the empty spots. </li>
  <li>Right column: Density plot of the foreground and background log
intensity.
    <ol type="a">
      <li>Foreground density plot: it should be composed of 2 peaks. A
smaller peak in the low intensity region containing missing spots and
negative control spots, and a higher one representing the rest of the
spots (probes). The number of present and absent spots, excluding empty
controls, estimated by EM algorithm is indicated on the graph. </li>
      <li>Background density plot: one peak in the low intensity
region. If a slide is of good quality, the background peak should not
overlap too much with the foreground peak corresponding to the bulk of
the data.</li>
    </ol>
  </li>
</ol>
Density plots are used to compare foreground and background peaks,
using the X-axis scale. They should be clearly separated. The number of
missing spots should be low. Missing spots ids may be incorporated in
the analysis later, e.g. by down weighting them in linear models.
<h4>Examples</h4>
This example uses 9-mer hybridization data performed in the Functional
Genomics Core Facility in UCSF. This print-run was created using Operon
Version 2 Mouse oligonucleotides.<br>
<br>
<font face="courier">
&gt; library(arrayQuality)<br>
&gt; datadir &lt;- system.file("data", package="arrayQuality")<br>
&gt; PRv9mers(fnames="12Mm250.gpr",path=datadir, prname="12Mm")<br>
</font>
<br>
<br>
<div align="center">
<img src="images/9mersDiagnosticPlot.png"
 alt="Example of 9-mers hybridization diagnostic plot" align="middle">
<p align="center">Figure 3: Example of diagnostic plot for 9-mers
hybridization</p>
</div>
<h2><a name="_Toc72663199">3.2 Quality Control
hybridizations</a><o:p>&nbsp;</o:p></h2>
<p class="MsoNormal"><br>
9-mers hybridizations help verify that oligonucleotides have been
spotted
properly on the slides. The next print-run quality control step will be:</p>
<p class="MsoNormal"><o:p>&nbsp;</o:p></p>
<p class="MsoNormal" style="margin-left: 0.75in; text-indent: -0.25in;"><span
 style=""><span style="">1.<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Detect any difference in overall signal intensity
compared to other print-runs</p>
<p class="MsoNormal" style="margin-left: 1.25in; text-indent: -0.25in;"><span
 style=""><span style="">a.<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>70-mers oligonucleotides hybridizations</p>
<p class="MsoNormal" style="margin-left: 1.25in; text-indent: -0.25in;"><span
 style=""><span style="">b.<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Selection of several test slides to ensure that
the same
quantity of material was spotted across the platter, as a print-run
will
generate 255 slides using the same well for one probe. QCHybs are
performed
using one slide from the beginning of the print, one from the middle,
one
from the end (e.g. numbers 20,100 and 255 in the Functional Genomics
Core
Facility).</p>
<p class="MsoNormal" style="margin-left: 0.5in;"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.75in; text-indent: -0.25in;"><span
 style=""><span style="">2.<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Check if the GAL file was generated properly, i.e.
check
that no error was made with ordering or orientation of the plates
during the
print.</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.75in; text-indent: -0.25in;"><span
 style=""><span style="">3.<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Reproducibility:</p>
<p class="MsoNormal" style="margin-left: 0.5in;">A good way to verify
the
quality of a new print is to hybridize known samples to new slides.
Then, we
can compare signal intensity from the new slides to existing data, and
check
that there is no loss in signal. Log ratios (M) for known samples
should be
similar across print-runs. Example of samples used for QCHybs includes
Human
Reference pool, Mouse liver, Mouse lung, with dye swaps.<br>
<br>
</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">The function in the package which performs the
quality
assessment for QCHybs is <span style="font-family: Courier;">PRvQCHyb()</span>.
<br>
</p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Copy the QCHybs gpr files from the <b>SAME
print-run</b>
(same GAL file) in a directory.</p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Change R working directory to the one containing
your
gpr files as described in section 1.</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Type:</p>
<p class="MsoNormal"
 style="margin-left: 0.25in; font-family: courier new;">&gt; <span
 class="codeChar">PRvQCHyb(prname="9Mm")</span></p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">where <span class="codeChar">prname</span> is the
name
of the print-run. For more details about its arguments, please refer to
the
online manual.<br>
&nbsp; <o:p><br>
<br>
</o:p></p>
<h3><a name="_Toc72663200">Results</a><o:p>&nbsp;</o:p></h3>
<p class="MsoNormal"><span style="font-family: Courier;">PRvQCHyb()</span>
returns a diagnostic plot as an image in .png format for each tested
slide.<br>
<br>
</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Throughout our document, we will be using the
color code
described in Table 1 to highlight control spots.</p>
<p class="MsoNormal" style="text-align: left;"><o:p>&nbsp;</o:p></p>
<div style="text-align: center;">
</div>
<br>
<br>
<table
 style="text-align: left; margin-left: auto; margin-right: auto; width: 200px;"
 border="1" cellpadding="2" cellspacing="2">
  <tbody>
    <tr align="left">
      <td style="vertical-align: top;">Positive controls<br>
      </td>
      <td style="vertical-align: top;">Red<br>
      </td>
    </tr>
    <tr align="left">
      <td style="vertical-align: top;">Empty controls<br>
      </td>
      <td style="vertical-align: top;">Blue<br>
      </td>
    </tr>
    <tr align="left">
      <td style="vertical-align: top;">Negative controls<br>
      </td>
      <td style="vertical-align: top;">Navy Blue<br>
      </td>
    </tr>
    <tr align="left">
      <td style="vertical-align: top;">Probes<br>
      </td>
      <td style="vertical-align: top;">Green<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: left;">Missing spots<br>
      </td>
      <td style="vertical-align: top; text-align: left;">White</td>
    </tr>
  </tbody>
</table>
<p class="MsoCaption" style="text-align: center;" align="center"><o:p></o:p><span
 style="">Table 1: Color code
used in arrayQuality<o:p></o:p></span></p>
<h3><br>
</h3>
<h3>Restrictions:<br>
</h3>
<p class="MsoNormal">Currently,<span class="codeChar"> PRQCHyb()</span>
supports Mouse genome (Mm) only. We will add Human data as soon as it
becomes
available.</p>
<p class="MsoNormal"><o:p>&nbsp;<br>
<br>
</o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<h3><a name="_Toc72663202">Description of the diagnostic plots</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Figure 4 shows an example of a nice print-run
QCHyb.<br>
<br>
</p>
<p class="MsoNormal"><o:p></o:p></p>
<ol style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">MA-plot of raw M values. No background
subtraction is performed. The colored lines represent the loess curves
for each print-tip group. The red dots highlight&nbsp; any spot with
corresponding weighted value&nbsp; less than 0. Users can create their
own weigthing scheme or function. Things to look for in a MA-plot are
saturation of spots and the trend of loess curves, which is an
indicator of the amount of normalization to be performed.<br>
  </li>
</ol>
<o:p></o:p>
<ol style="margin-top: 0in;" type="1">
</ol>
<ol start="2" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Boxplot of raw M values by print-tip
group, without background subtraction.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="3" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Spatial plot of rank of raw M values
(no background subtraction): Each spot is ranked according to its M
value. We use a blue to yellow color scale,where blue represents the
higher rank (1), and yellow represents the lower one. Missing spots are
represented as white squares.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="4" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Spatial plot of A values. The color
indicates the strength of the signal intensity, i.e. the darker the
color, the stronger the signal. Missing spots are represented in white.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="5" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Histogram of the signal-to-noise
log-ratio (SNR) for Cy5 and Cy3 channels. The mean and the variance of
the signal are printed on top of the histogram. In addition, overlay
density of SNR stratified by different control types (status) are
highlighted. Their color schemes are provided in Table 1. The SNR is a
good indicator for dye problems. The negative controls and empty
controls density lines should be closer, almost superimposed.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="6" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Comparison of Mvalues of probes known
to be differentially expressed from the tested array to average Mvalues
obtained during previous hybridizations. This plot is aimed at
verifying the reproducibility of print-runs. The dotted lines are the
diagonal (no change) and the +2/-2 fold change lines. Each probe is
represented by a number, and described in the file MmDEGenes.xls. Most
of the spots should lie between the +2/-2 fold-change regions. If the
technique was perfect, you should see a straight line on the diagonal.
If any probe falls off this region (number 29 here), you can look up
its number in our probe list in MmDEgenes.xls and get more information
about it.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="7" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Dot plot of controls A values, without
background subtraction. Controls with more than 3 replicates are
represented on the Y-axis, the color scheme is represented in Table 1.
Intensity of positive controls should be in the high-intensity region,
negative and empty controls should be in the lower intensity region.
Positive controls range and negative/empty controls range should be
separated. Replicate spots signal should be tight.<br>
  </li>
</ol>
<p class="MsoNormal"><o:p>&nbsp;</o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<h3><a name="_Toc72663203">Example</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Data for this example was provided by the
Functional
Genomics Core Facility in UCSF. We have tested slide number 137 from
print-run 9Mm. This print-run uses Operon Version 2 Mouse oligos.
Results are
represented figure 4.<br>
<o:p>&nbsp;</o:p></p>
<p class="code" style="font-family: courier new;">&gt;
library(arrayQuality)</p>
<p class="code" style="font-family: courier new;">&gt; datadir &lt;-
system.file("data", package="arrayQuality")</p>
<p class="code" style="font-family: courier new;">&gt;
PRvQCHyb(fnames=&#8221;9Mm137.png&#8221;, path=datadir, prname="9Mm")</p>
<p class="MsoNormal"><o:p></o:p></p>
<div style="text-align: center;">
<o:p></o:p><img src="images/QCHybDiagPlot.png" title=""
 alt="Example of QCHyb diagnostic plot"
 style="width: 800px; height: 500px;"><o:p><br>
</o:p></div>
<p class="MsoCaption" style="text-align: center;" align="center">Figure
4:
Diagnostic plot for print-run Quality Control hybridization</p>
<o:p></o:p>
<p class="MsoNormal"><o:p></o:p><span style="font-weight: bold;"><br>
<br>
</span></p>
<h1 style="margin-left: 0in; text-indent: 0in;"><a name="_Toc72663205">4.
General hybridization quality</a><br>
</h1>
<p class="MsoNormal"><o:p><br>
&nbsp;<br>
</o:p></p>
<p class="MsoNormal">This component is aimed at verifying the
performance of
your hybridization, given the good quality of the slide, before any
preprocessing steps or further quality assessment on individual spots.
This
is where you determine if your experiment quality is good enough to
enter
your dataset. For example, you will need to remove any hybridization
with
very low SNR, or large spatial artifacts.<br>
<br>
</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Our package provides two kinds of quality control
plots.
The first one is a qualitative quality control measurement as a
diagnostic
plot. It is a quick visual way to determine hybridization quality
gathering
information from several statistical tools. More details on individual
diagnostic plots can be found in the vignette &#8220;marrayPlots&#8221; in
the package <span style="font-family: courier new;">marray</span>. The
second
one is a more quantitative comparison of slide quality. We extract some
statistical measures from the test slide and we compare them against
results
obtained for a collection of slides of &#8220;good quality&#8221; to assess
the quality of the hybridization. This comparison is visualized through
a
comparative boxplot. Results are displayed in a HTML report. Figure 5
shows a
screen shot of a typical HTML report. Users can click on each image to
obtain
a higher resolution plot.<br>
</p>
<p class="MsoNormal">Diagnostic plots can be generated&nbsp; for
different image processing software format: GenePix format files (.gpr
files), Spot format files (.spot) and Agilent format files, or from <span
 style="font-family: courier new,courier,monospace;">marrayRaw</span>
or <span style="font-family: courier new,courier,monospace;">RGList</span>
objects. Most arguments can also be customized to match your own data:
which probes are used as controls, which column of the image processing
output file is used
to define your spot types... You can also specify your own collection
of good quality
slides using the functions <span class="codeChar"
 style="font-family: courier new;">globalQuality</span><span
 style="font-family: courier new;"> </span>and <span class="codeChar"
 style="font-family: courier new;">qualRefTable</span>. For more
details about
these functions, please refer to the online help and the example at the
end
of this Section.<o:p><br
 style="font-family: courier new,courier,monospace;">
</o:p></p>
<h3><br>
</h3>
<h3>To generate quality plots: gpr files: <span
 style="font-family: courier new,courier,monospace;">gpQuality()</span></h3>
We provide 3 main functions to generate quality plots:
<b><font face="courier">gpQuality(), spotQuality() and agQuality()</font></b>.
We will use gpQuality as an example, but the following can be directly
applied to spotQuality or agQuality.<span
 style="font-family: courier new,courier,monospace;"> gpQuality()</span>will
generate both diagnostic plots and comparative boxplots. It uses by
default spot types from the Functional Genomics Core Facility in UCSF.
To use your own spot types, please refer to the end of this Section.<br>
<br>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Copy the gpr files from the <b style="">SAME
print-run</b> (same GAL file) in a directory.</p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Change R working directory to the one containing
your
gpr files as described in Section 1<br>
</p>
<p class="MsoNormal" style="margin-left: -0.25in; text-indent: 0.5in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>To generate<span style="font-weight: bold;"> both
diagnostic plots and comparative boxplots</span> on all files in the
directory, run:<br>
<span class="codeChar" style="font-family: courier new;">&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp; <br>
&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &gt; result &lt;-
gpQuality(organism=&#8221;Mm&#8221;)<br>
<span style="font-family: times new roman,times,serif;"><br
 style="font-family: times new roman,times,serif;">
</span>&nbsp; </span><span style=""><span style="">&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp; -<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>To generate<span style="font-weight: bold;">
diagnostic
plots only</span>, run:<br>
&nbsp;
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp;&nbsp; <span
 style="font-family: courier new;"><br>
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp; &gt;&nbsp; result &lt;-
gpQuality(organism="Mm", compBoxplot="FALSE")<br>
</span></p>
<p class="MsoNormal" style="margin-left: -0.25in; text-indent: 0.5in;"><span
 style="font-family: courier new;"><span
 style="font-family: times new roman,times,serif;">In this case,
quantitative
quality measures will not be calculated and the HTML report will <span
 style="font-weight: bold;">not</span> be generated.<br>
</span><br>
</span></p>
<p class="MsoNormal" style="margin-left: -0.25in; text-indent: 0.5in;">-
To
write down your <span style="font-weight: bold;">quantitative quality
measures and your normalized data to a file</span>: set <span
 style="font-family: courier new;">output = TRUE</span> when calling <span
 style="font-family: courier new;">gpQuality</span>:</p>
<span style="font-family: courier new;">&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;
&gt; result &lt;- gpQuality(organism="Mm",
output=TRUE)</span><br>
<br>
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp; This command will create
two files: quality.txt, which contains your quality measures, and
NormalizedData.xls, which contains your normalized M values. If you
have set
<span style="font-family: courier new;">compBoxplot =
FALSE</span>, quantitative quality
measures are not calculated. Therefore, you will not generate the
quality.txt
file.<br>
&nbsp;<br>
<h3><br>
</h3>
<h3>To generate quality plots from <span
 style="font-family: courier new,courier,monospace;">marrayRaw/RGList</span>
objects: <span style="font-family: courier new,courier,monospace;">maQualityPlots</span></h3>
<span style="font-family: courier new,courier,monospace;"><span
 style="font-family: times new roman,times,serif;">This function can be
use to obtain quality plots for data generated with other image
processing software, like Spot for example. </span>maQualityPlots()</span>
will generate diagnostic plots only. It uses the spot types defined
when creating the R object. To learn more about how to read data into a
<span style="font-family: courier new,courier,monospace;">marrayRaw</span>
or a <span style="font-family: courier new,courier,monospace;">RGList<span
 style="font-family: times new roman,times,serif;"> </span></span>object,
please refer to <span
 style="font-weight: bold; font-family: courier new,courier,monospace;">marray
</span>or <span
 style="font-family: courier new,courier,monospace; font-weight: bold;">limma</span><span
 style="font-weight: bold;"> </span>packages Vignettes.<br>
<br>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">
&nbsp;&nbsp; &nbsp; &nbsp;&nbsp;&nbsp; </span></span></span>To
generate diagnostic plots: if <span
 style="font-family: courier new,courier,monospace;">rawdata </span>is
your <span style="font-family: courier new,courier,monospace;">marrayRaw/RGList</span>
object, type:</p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style="font-family: courier new,courier,monospace;">&gt;
maQualityPlots(rawdata) </span><br>
</p>
<br>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<h3><a name="_Toc72663207">Results</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal"><span class="codeChar"
 style="font-family: courier new;">gpQuality()</span> outputs<br>
</p>
<ul>
  <li>In the working directory:</li>
</ul>
<p class="MsoNormal" style="margin-left: 39pt; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>two plots for each test slide (a diagnostic plot
and a
comparative boxplot)</p>
<p class="MsoNormal" style="margin-left: 39pt; text-indent: -0.25in;"><span
 style=""><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>a HTML quality report<br>
<br>
</p>
<ul>
  <li>In the R global environment, the function returns a list of two
objects:<span style="font-family: Wingdings;"><span style=""><span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;
    <br>
    </span></span></span></li>
</ul>
<p>&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; -&nbsp;&nbsp;&nbsp; A
marrayRaw
object describing all tested slides</p>
<p>&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; -&nbsp;&nbsp;&nbsp; A quality
measures matrix: this matrix contains all comparison measures values
extracted for each test slide. Each column of the matrix represents a
different slide.<br>
</p>
<ul style="margin-left: 40px;">
</ul>
<p class="MsoNormal" style="margin-left: 3pt;"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 3pt;">For each slide, you will
find
on the report how many of your slide&#8217;s results are below the
recommended range. If you want to specify a directory to store the
results,
you can do it by modifying the argument <span class="codeChar"
 style="font-family: courier new,courier,monospace;">resdir</span><span
 style="font-family: courier new,courier,monospace;"> </span>accordingly.
For
more details about <span class="codeChar"
 style="font-family: courier new;">gpQuality</span> arguments, please
refer to
the online manual.<br>
</p>
<p class="MsoNormal" style="margin-left: 3pt;"><br>
</p>
<p class="MsoNormal" style="margin-left: 3pt; text-align: center;"><img
 src="images/htmlReport.png" title=""
 alt="Example ofgpQuality HTML report"
 style="width: 600px; height: 445px;"><br>
Figure 5: Example of
HTML report generated by gpQuality<br>
</p>
<span style="font-family: courier new,courier,monospace;">maQualityPlots()</span>
output:
<ul>
  <li>In the working directory:</li>
</ul>
<span style=""><span style="">&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; -<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>one diagnostic plot for each test slide <br>
<a name="_Toc72663206"></a>
<h3><br>
</h3>
<h3><a name="_Toc72663206">Restrictions</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal"><span style="font-family: courier new;">gpQuality</span>
calls two key functions, <span style="font-family: courier new;">maQualityPlots
</span>and <span style="font-family: courier new;">qualBoxplot</span>.
<span style="font-family: courier new;">qualBoxplot</span> supports
Mouse (Mm) and
Human (Hs) genomes only. To generate quality plots for other genomes,
you
need to set <span style="font-family: courier new;">gpQuality </span>argument
<span style="font-family: courier new;">compBoxplot = FALSE</span>. In
this
case, only the diagnostic plots will be generated.<br>
</p>
<p class="MsoNormal"><o:p>&nbsp;</o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<h3><b style="">Description of the diagnostic plots:<o:p></o:p></b></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Figure 6 represents an example of a good
hybridization
diagnostic plot.</p>
<p class="MsoNormal"><o:p></o:p></p>
<ol style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">MA-plot of raw M. No background
subtraction is performed. The colored lines represent the loess curves
for each print-tip group. The red dots highlight&nbsp; any spot with
corresponding weighted value&nbsp; less than 0. Users can create their
own weigthing scheme or function. Things to look for in a MA-plot are
saturation of spots and the trend of loess curves, which is an
indicator of the amount of normalization to be performed.</li>
</ol>
<p class="MsoNormal" style="margin-left: 0.25in;"><o:p></o:p></p>
<ol start="2" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">MA-plot of normalized data density. By
default, print-tip loess normalization is used. Instead of the typical
MA-plot, we have used the package "<span
 style="font-family: courier new;">hexbin</span>" to highlight density
of dots on the MA-plot. A light yellow color indicates a high density
of dots, whereas blue color represents a lower density. This plot gives
you information on the bulk of your data intensity (low/high signal)</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="3" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Spatial plot of rank of raw M values
(no background subtraction): Each spot is ranked according to its M
value. We use a blue to yellow color scale,where blue represents the
higher rank (1), and yellow represents the lower one. Missing spots are
represented as white squares. This is a quick way to visually detect
uneven hybridization and missing spots.</li>
</ol>
<p class="MsoNormal" style="margin-left: 0.25in;"><o:p></o:p></p>
<ol start="4" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Spatial plot of normalized M values
ranks. By default, print-tip loess normalization is used. Each spot is
ranked according to its M value. We use a blue to yellow color
scale,where blue represents the higher rank (1), and yellow represents
the lower one. Missing spots are represented as white squares. In
addition, flagged spots are higllighted by a black square. This type of
graphical representation helps verify that normalization removed any
spatial effects.</li>
</ol>
<p class="MsoNormal" style="margin-left: 0.25in;"><o:p></o:p></p>
<ol start="5" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Spatial plot of raw A values. The
color indicates the strength of the signal intensity, i.e. the darker
the color, the stronger the signal. Missing spots are represented in
white.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="6" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Histogram of the signal-to-noise
log-ratio (SNR) for Cy5 and Cy3 channels. The mean and the variance of
the signal are printed on top of the histogram. In addition, overlay
density of SNR stratified by different control types (status) are
highlighted. Their color schemes are provided in Table 1. The SNR is a
good indicator for dye problems. The negative and empty controls
density lines should be closer, almost superimposed.</li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="7" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Dot plot of controls normalized M
values. Controls with more than 3 replicates are represented on the
Y-axis, the color scheme is represented in Table 1. Controls M values
should be tight. and close to 0.<br>
  </li>
</ol>
<p class="MsoNormal"><o:p></o:p></p>
<ol start="8" style="margin-top: 0in;" type="1">
  <li class="MsoNormal" style="">Dot plot of controls A values, without
background subtraction. Controls with more than 3 replicates are
represented on the Y-axis, the color scheme is represented in Table 1.
Intensity of positive controls should be in the high-intensity region,
negative and empty controls should be in the lower intensity region.
Positive controls range and negative/empty controls range should be
separated. <br>
  </li>
</ol>
<br>
<h3><a name="_Toc72663208">Description of the comparative Boxplot:</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Figure 7 shows an example of a comparative boxplot.</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">We have chosen a wide range of measures to
quantify the
quality of a typical hybridization: single channel measures (range of
foreground signal, MAD of background, signal to noise ratio&#8230;), two
channel measures (median A values for each type of controls, amount of
normalization needed&#8230;), percentage of flagged spots... Some measures
have been negated such that the quality scale had an increasing trend
from
problematic to good quality. <br>
</p>
<p class="MsoNormal">For each measure, we have represented the
following on
the graph :<br>
</p>
<p class="MsoNormal" style="margin-left: 0.25in;"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style="font-family: &quot;Times New Roman&quot;;"><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Boxplot of the reference slides values.</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style="font-family: &quot;Times New Roman&quot;;"><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>1<sup>st</sup> and 3<sup>rd</sup> quantiles before
scaling for each boxplot.</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal" style="margin-left: 0.5in; text-indent: -0.25in;"><span
 style="font-family: &quot;Times New Roman&quot;;"><span style="">-<span
 style="font-family: &quot;Times New Roman&quot;; font-style: normal; font-variant: normal; font-weight: normal; font-size: 7pt; line-height: normal; font-stretch: normal; font-size-adjust: none;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span>Y-axis on the right : for each measure, we have
printed
2 values. The first one is the percentage of reference slides measures
under
your slide&#8217;s result. The second one is your slide value for this
measure before scaling.<br>
</p>
&nbsp;&nbsp;&nbsp; &nbsp; -&nbsp;&nbsp;&nbsp; We have scaled all the
results
to be able to compare them on the same graph. &nbsp;
<p class="MsoNormal"><o:p></o:p></p>
&nbsp;&nbsp;&nbsp; &nbsp; -&nbsp;&nbsp; The red dots are the test slide
scaled values<br>
<br>
&nbsp;<br>
The 16 measures we have selected are listed below. <br>
<br>
<p class="MsoNormal" style="margin-left: 40px;">1. <span
 style="font-weight: bold;">rangeRf</span>: Range of Cy5 foreground,
where the
range is defined by:<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">rangeRf = max(log2
(median
Cy5 foreground)) - min(log2(median Cy5 foreground))<br>
where median Cy5 foreground corresponds to the "F635 Median" column of
the
gpr file.<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><br>
2. <span style="font-weight: bold;">rangeGf</span>: Range of Cy3
foreground, where the
range is defined by:<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">rangeGf = max(log2
(median
Cy3 foreground)) - min(log2(median Cy3 foreground))<br>
where median Cy3 foreground corresponds to the "F532 Median" column of
the
gpr file.</p>
<p class="MsoNormal" style="margin-left: 40px;"><br>
3.<b> -RbMad: </b>Cy5 background MAD<b><br>
</b></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">RbMad = mad[log2(Cy5
background)]</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- Cy5 background corresponds to the "B635 Median" column of the gpr file<br>
- MAD = <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">median{
| Y
&#8211;mu | }, when Y is normal<br>
<br>
</span></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><span style="" lang="DA">4.&nbsp;
-<b>GbMad: </b></span>Cy3 background MAD<b><br>
</b></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">GbMad = mad[log2(Cy3
background)]</p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
where:<br>
- Cy3 background corresponds to the "B532 Median" column of the gpr file<br>
- MAD = <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">median{
| Y
&#8211;mu | }, when Y is normal<br>
</span>
<p><br>
</p>
5.&nbsp; <span style="font-weight: bold;">Median</span> <b>RS2N:</b>
Median
Signal To Noise log-ratio for Cy5<br>
<p class="MsoNormal">RS2N = log2( mean Cy5 foreground / Median Cy5
background
)<br>
</p>
<p class="MsoNormal">RS2Nmedian = median(RS2N)&nbsp;</p>
<p class="MsoNormal"><o:p>where:<br>
- mean Cy5 foreground is the "F635 Mean" column of the gpr file<br>
- median Cy5 background is the "B635 Median" column of the gpr file<br>
&nbsp;</o:p></p>
<p class="MsoNormal">6. <b>Median GS2N: </b>Median Signal To Noise
log-ratio
for Cy3</p>
<p class="MsoNormal">GS2N = log2( mean Cy3 foreground / Median Cy3
background
)<br>
</p>
<p class="MsoNormal">GS2Nmedian = median(GS2N)<br>
</p>
<p class="MsoNormal">where:<br>
- mean Cy3 foreground is the "F532 Mean" column of the gpr file<br>
- median Cy3 background is the "B532 Median" column of the gpr file<br>
</p>
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><br>
7. <span style="font-weight: bold;">-Median A for empty control</span>:<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">Median A = [
log2(median Cy5
foreground) + log2(Cy3 foreground) ] / 2</p>
<p class="MsoNormal" style="margin-left: 40px;">Median A for empty
control =
median( A(Empty controls))</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- median Cy5 foreground corresponds to the "F635 Median" column of the
gpr
file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
- Empty controls are the probes labelled "Empty" <br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p>&nbsp;</o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">8. <b>-Median A for
negative
control</b>:<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">Median A = [
log2(median Cy5
foreground) + log2(Cy3 foreground) ] / 2</p>
<p class="MsoNormal" style="margin-left: 40px;">Median A for negative
control
= median( A(Negativecontrols))</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- median Cy5 foreground corresponds to the "F635 Median" column of the
gpr
file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
- Negative controls are the probes labelled "Negative" <br>
<br>
</p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">9.&nbsp;<b>Median A
values
for Positive controls:</b>&nbsp;</p>
<p class="MsoNormal" style="margin-left: 40px;"><b><o:p></o:p></b></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">A = [ log2(median Cy5
foreground) + log2(Cy3 foreground) ] / 2</p>
<p class="MsoNormal" style="margin-left: 40px;">Median A for positive
control
= median( A(Positive controls))</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- median Cy5 foreground corresponds to the "F635 Median" column of the
gpr
file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
- Positive controls are the probes labelled "Positive" <br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p><br>
</o:p></p>
<p class="MsoNormal" style="margin-left: 40px;">10. <span
 style="font-weight: bold;">Difference between A values for Positive
controls
and&nbsp; A values for Negative controls</span></p>
<div style="margin-left: 40px; font-weight: bold;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p>difference =&nbsp;
</o:p>median( A(Positive controls)) - median( A(Negativecontrols))<o:p><br>
<br>
</o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">11. <b>-varRepA</b>:
variance
of replicates spots A values <br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">varRepA = var[
A(replicates)
]<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- A = [ log2(median Cy5 foreground) + log2(Cy3 foreground) ] / 2<br>
- median Cy5 foreground corresponds to the "F635 Median" column of the
gpr
file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
<o:p><br>
<br>
</o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">12. <b>-msePtip</b>:
MSE of M
values by print-tip group, no background subtraction</p>
<p class="MsoNormal" style="margin-left: 40px;">&nbsp;M =&nbsp;
log2(Median
Cy5 foreground) - log2(Median Cy3 foreground)<br>
msePtip = MSE( mean M by print-tip)<o:p><br>
</o:p></p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p>where:<br>
</o:p>- median Cy5 foreground corresponds to the "F635 Median" column
of the
gpr file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
- <span style="font-size: 9.5pt;">MSE(X) = <a
 href="http://stat-www.berkeley.edu/users/stark/SticiGui/Text/#expectation">E</a></span>(<span
 style="font-size: 9.5pt;"> (X-<em><span style="font-family: Arial;">t</span></em>)<sup>2</sup>
</span>), with t a
parameter and X an estimator of t.<br>
<o:p>&nbsp;</o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">13. <b>-mseFit</b>:
MSE of
lowess curve</p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">fit = lowess(A,
M)<o:p>&nbsp;</o:p><br>
mseFit = MSE(fit$y)<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;"><o:p>where:<br>
- </o:p>A = [ log2(median Cy5 foreground) + log2(Cy3 foreground) ] / 2<br>
- M =&nbsp; log2(Median Cy5 foreground) - log2(Median Cy3 foreground)<br>
<o:p></o:p>- median Cy5 foreground corresponds to the "F635 Median"
column of
the gpr file<br>
- median Cy3 foreground corresponds to the "F532 Median" column of the
gpr
file<br>
- <span style="font-size: 9.5pt;">MSE(X) = <a
 href="http://stat-www.berkeley.edu/users/stark/SticiGui/Text/#expectation">E</a></span>(<span
 style="font-size: 9.5pt;"> (X-<em><span style="font-family: Arial;">t</span></em>)<sup>2</sup>
</span>), with t a
parameter and X an estimator of t.<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><br>
14. <span style="font-weight: bold;">-Percentage of flagged spots</span><br
 style="font-weight: bold;">
<br>
[number of spot with flag &lt; 0 / number of spots] * 100</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">where flag is the
information
from the "Flags" column of the gpr file. Only spots with flag less than
0 are
taken into account.<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p>&nbsp;</o:p><span
 style="" lang="DA"><o:p></o:p></span><br>
15. <b>-M values MMRmad<br>
</b></p>
<p class="MsoNormal" style="margin-left: 40px;">MMR = Mmean &#8211;
Mmedian<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">MAD(MMR)<br>
<br>
where:<br>
- Mmean =&nbsp; log2(Mean Cy5 foreground) - log2(Mean Cy3 foreground)<br>
M values calculated using mean signal<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">- Mmedian = log2(Median
Cy5
foreground) - log2(Median Cy3 foreground)<br>
M values calculated using median signal<br>
</p>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">- mean Cy5 (Cy3)
foreground
is the "F635 Mean" ("F532 Mean") column of the gpr file <br>
- median Cy5 (Cy3) foreground is the "F635 Median" ("F532
Median")&nbsp;
column of the gpr file<br>
- MAD = <span style="font-size: 12pt; font-family: &quot;Times New Roman&quot;;">median{
| Y
&#8211;mu | }, when Y is normal<br>
</span><o:p>&nbsp;</o:p></p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;">16. <span
 style="font-weight: bold;">-Percentage of spots with abs[MMR] &gt;
0.5</span><br style="font-weight: bold;">
</p>
<p class="MsoNormal" style="margin-left: 40px;">where:<br>
- MMR = Mmean &#8211; Mmedian</p>
<p class="MsoNormal" style="margin-left: 40px;">- Mmean =&nbsp;
log2(Mean Cy5
foreground) - log2(Mean Cy3 foreground)<br>
M values calculated using mean signal<br>
</p>
<p class="MsoNormal" style="margin-left: 40px;">- Mmedian = log2(Median
Cy5
foreground) - log2(Median Cy3 foreground)<br>
M values calculated using median signal<br>
</p>
<div style="margin-left: 40px;">
</div>
<div style="margin-left: 40px;">
- mean Cy5 (Cy3) foreground is the "F635 Mean" ("F532 Mean") column of
the
gpr file <br>
- median Cy5 (Cy3) foreground is the "F635 Median" ("F532
Median")&nbsp;
column of the gpr file<br>
<br>
</div>
<div style="margin-left: 40px;">
</div>
<p class="MsoNormal" style="margin-left: 40px;"><o:p></o:p></p>
<p class="MsoNormal"><o:p></o:p></p>
<h3><a name="_Toc72663209">Example</a></h3>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal">Data for this example was provided by the
Functional
Genomics Core Facility in UCSF. We have tested slide number "137" from
print-run "9Mm". This array was fabricated using Operon Version 2 Mouse
oligos and the hybridization measures differential gene expression in
two RNA
samples, Mouse Liver and Mouse Reference Pool. Results are represented
Figure
5 and Figure 6.<br>
<br>
</p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="code" style="font-family: times new roman,times,serif;">To
generate
diagnostic plots, comparative boxplots, HTML report and to write your
quality
measure and normalized data to a file in a directory named "Results":</p>
<p class="code" style="font-family: courier new;">&gt;
library(arrayQuality)</p>
<p class="code" style="font-family: courier new;">&gt; datadir &lt;-
system.file("data", package="arrayQuality")</p>
<p class="code" style="text-align: left; font-family: courier new;"
 align="left">&gt; result &lt;-
gpQuality(<span style="color: black;">fnames = "9Mm137.gpr", path =
datadir,<o:p></o:p>organism = &#8221;Mm&#8221;, output = TRUE, resdir =
"Results")<o:p></o:p></span></p>
<p class="MsoNormal"><o:p></o:p></p>
<p class="MsoNormal"><o:p><br>
</o:p></p>
<div style="text-align: center;">
<img src="images/genHybDiagPlot.png" title=""
 alt="Example of general hybridization diagnostic plot"
 style="width: 576px; height: 432px;"><o:p><br>
&nbsp;<br>
</o:p>Figure 6: General hybridization quality diagnostic plot<br>
<div style="text-align: left;">
<br>
<br>
<div style="text-align: center;">
<img src="images/comparativeBoxplot.png" title=""
 alt="General hybridization quality: comparative boxplot"
 style="width: 500px; height: 375px;"><br>
</div>
</div>
</div>
<p class="MsoCaption" style="text-align: center;" align="center">Figure
7:
Comparative boxplot</p>
<p class="MsoNormal"><o:p></o:p></p>
<br>
<br>
<br>
<br>
<h3>Customizing gpQuality</h3>
<span style="font-family: courier new,courier,monospace;"><br>
</span><span style="font-family: courier new,courier,monospace;">arrayQuality
</span>is currently using look-up tables adapted to hybridizations
performed
in the Functional Genomics Core Facility in UCSF.&nbsp; Depending on
your
data, you may find that the probes defined as controls in <span
 style="font-family: courier new,courier,monospace;">arrayQuality </span>are
not present on your array, leading to NAs in the comparative boxplot,
or you
may be working with a genome for which we are not providing references<span
 style="font-family: courier new,courier,monospace;"></span><span
 style="font-family: courier new,courier,monospace;">. gpQuality </span>has
several arguments that you can modify <span
 style="font-family: times new roman,times,serif;">in order</span><span
 style="font-family: courier new,courier,monospace;"><span
 style="font-family: times new roman,times,serif;">
to use your own spot types or your own collection of good slides.
gpQuality arguments are listed below:</span></span><br>
<span style="font-family: courier new,courier,monospace;"><br>
gpQuality(fnames = NULL, path = ".", organism = c("Mm", "Hs"),<br>
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp; compBoxplot = TRUE,
reference = NULL, <br>
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; &nbsp; controlMatrix =
controlCode, controlId = c("ID", "Name"), <br>
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; output = FALSE,
resdir =".", dev= "png", DEBUG = FALSE,...)<br>
<span style="font-family: times new roman,times,serif;"><br>
<br>
To use your own set of spot types (i.e. controls...): you will need to
change <span style="font-family: courier new,courier,monospace;">controlMatrix</span>
and/or <span style="font-family: courier new,courier,monospace;">controlId</span>.<br>
<br>
To use your own collection of good slides: you will need to modify <span
 style="font-family: courier new,courier,monospace;">reference</span>.</span></span><span
 style="font-family: courier new,courier,monospace;"><br>
</span><br>
<span style="font-weight: bold;">To use your own set of spot types:</span><br
 style="font-weight: bold;">
<br>
The spot types used in arrayQuality are defined in a 2 column matrix
called
controlCode. <br>
<br>
<table
 style="text-align: left; width: 300px; margin-left: auto; margin-right: auto;"
 border="1" cellpadding="2" cellspacing="2">
  <tbody>
    <tr>
      <td style="vertical-align: top;">Pattern<br>
      </td>
      <td style="vertical-align: top;">Name<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">Buffer<br>
      </td>
      <td style="vertical-align: top;">Buffer<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">Empty<br>
      </td>
      <td style="vertical-align: top;">Empty<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">EMPTY<br>
      </td>
      <td style="vertical-align: top;">Empty<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">AT<br>
      </td>
      <td style="vertical-align: top;">Negative<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">M200009348<br>
      </td>
      <td style="vertical-align: top;">Positive<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">M200003425<br>
      </td>
      <td style="vertical-align: top;">Positive<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top;">NLG<br>
      </td>
      <td style="vertical-align: top;">con<br>
      </td>
    </tr>
  </tbody>
</table>
<br>
<div style="text-align: center;">
Table2: Examples of controls used in arrayQuality<br>
</div>
<br>
<br>
To define your own spot types, you will need to replace the default
values in controlCode with your values. The easiest way to do it is to
create a&nbsp; tab-delimited text file named SpotTypes.txt, and read it
into arrayQuality using the function readcontrolCode. It is also
possible to create a new controlCode matrix directly.<br>
<br>
1. If you want to use a Spot Types file:<br>
<br>
A spot types files is a tab-delimited text file which allows you to
identify different types of spots from the gene list. It should contain
at least a column named SpotType where all different spot types are
listed and one or more other columns, which should have the same names
as columns in the GAL file, containing patterns or regular expressions
sufficient to identify the spot-type. For more information, you can
refer to the <span style="font-family: courier new,courier,monospace;">limma</span>
package userguide. <br>
<br>
<span style="font-weight: bold;">Warning:</span> You will need to
include a spot type named <span style="font-weight: bold;">probes</span>!!<br>
<br>
Below is an example of spot types files for the swirl dataset. In this
case there are only two types of spots, probes and controls.<br>
<br>
<div style="text-align: center;"><img src="images/mySpotTypes.png"
 title="" alt="example of spot types file"
 style="width: 417px; height: 214px;"><br>
</div>
<div style="text-align: center;"><br>
</div>
<div style="text-align: center;">Example of spot types file<br>
</div>
<br>
<br>
To read the new spot types in arrayQuality:<br>
<br>
- Create your spot types file.<br>
- Find which column of the file contains probes identification for each
type. In the example Figure 8, it is the "ID" column. You will need to
pass this column name as argument at the next step.<br>
- Read the spot types files using the readcontrolCode function. <br>
<p class="MsoNormal"
 style="text-align: justify; font-family: courier new,courier,monospace;">&gt;
controlCode &lt;-
readcontrolCode(file=&#8221;mySpotTypes.txt&#8221;, controlID="ID")<br>
</p>
<p class="MsoNormal" style="text-align: justify;">- Find which column
of the gpr file can be used to identify your new spot types. It is
typically the "ID" or the "Name" column.<br>
- To generate both types of plots: call <span
 style="font-family: courier new,courier,monospace;">gpQuality</span>
specifying your new controlCode matrix in <span
 style="font-family: courier new,courier,monospace;">controlMatrix</span>
and which column is used to define your spot types in <span
 style="font-family: courier new,courier,monospace;">controlID</span>.<br>
</p>
<p class="MsoNormal"
 style="text-align: justify; font-family: courier new,courier,monospace;">&gt;
result &lt;- gpQuality(controlMatrix = controlCode, controlId=&#8221;Id&#8221;)</p>
<br>
2. If you want to create a new <span
 style="font-family: courier new,courier,monospace;">controlCode</span>
matrix directly<br>
<br>
You will need to create another <span
 style="font-family: courier new,courier,monospace;">controlCode </span>table
containing two columns as well, and then overwrite the default <span
 style="font-family: courier new,courier,monospace;">controlCode </span>loaded
with <span style="font-family: courier new,courier,monospace;">arrayQuality</span>.<br>
&nbsp;&nbsp;&nbsp; - A column named "Pattern" containing your control
IDs<br>
&nbsp;&nbsp;&nbsp; - A column named "Name", describing what king of
control
is each probe (in particular what are Positive, Negative, Empty
controls)<br>
<br>
You can do it by creating a tab delimited text file and read it in R
after
loading <span style="font-family: courier new,courier,monospace;">arrayQuality</span>:<br>
<br>
<span style="font-family: courier new,courier,monospace;">&gt;
library(arrayQuality)<br>
&gt; mycontrolCode &lt;- as.matrix(read.table("mycontrolCode.txt",
sep="\t",</span><br style="font-family: courier new,courier,monospace;">
<span style="font-family: courier new,courier,monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
header=TRUE, quote="\"", fill=TRUE)))<br>
<br>
<span style="font-family: times new roman,times,serif;">Then, pass your
new matrix as argument when calling gpQuality. You can specify which
column of the gpr file contains probes identifiers in the controlID
arguments (typically, it would be "Id" or "Name").</span><br>
&gt; results &lt;- gpQuality(controlMatrix = mycontrolCode, controlID =
"ID")<br style="font-family: courier new,courier,monospace;">
</span><br>
<br>
<span style="font-weight: bold;">To use your own reference slides:</span><br>
<br>
&nbsp;If you would like to use your own set of reference slides, you
will
need to follow a few steps to create the necessary look-up tables. This
feature can be used for example if you want to study hybridization
quality
for other genomes, or if you would like to compare slide quality within
a a
large dataset. To generate your own references:<br>
<br>
&nbsp;&nbsp;&nbsp; 1. Gather the slides of "good" quality you would
like to
use as reference in a directory, for example "MyReferences". Slides can
be
from different print-runs.<br>
<br>
&nbsp;&nbsp;&nbsp; 2. Change R working directory to "MyReferences", as
described in Section 1.<br>
<br>
&nbsp;&nbsp;&nbsp; 3. Load arrayQuality package by typing <span
 style="font-family: courier new;">library(arrayQuality)</span> in R<br>
<br>
&nbsp;&nbsp;&nbsp; 4. Create your reference quality measures by typing:<br>
&nbsp;&nbsp;&nbsp; <span style="font-family: courier new;">&gt;
myReference
&lt;-&nbsp; globalQuality()</span><br style="font-family: courier new;">
<br>
&nbsp;&nbsp;&nbsp; 5. Change R working directory to the directory
containing
slides&nbsp; you would like to test, as described in Section 1. You can
only
compare slides from the same print-run here. If you have an experiment
&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp; using two print-runs, you will
need to
run gpQuality two times.<br>
<br>
&nbsp;&nbsp;&nbsp; 6. Run gpQuality using the reference measures and
the
scaling table you have generated:<br>
&nbsp;&nbsp;&nbsp; <span style="font-family: courier new;">&gt;
results &lt;-
gpQuality(reference = myReference)</span><br>
<br>
&nbsp;&nbsp;&nbsp; Other <span style="font-family: courier new;">gpQuality
</span>arguments described above can also be applied here.<br>
<p class="MsoCaption" style="text-align: left;"><br>
</p>
</body>
</html>
